homeassistant/components/fritz/common.py

```diff
@@ -111,16 +132,39 @@ async def async_start(self, options):
         )
 
     @callback
-    def async_unload(self):
+    def async_unload(self) -> None:
         """Unload FritzboxTools class."""
         _LOGGER.debug("Unloading FRITZ!Box router integration")
         if self._cancel_scan is not None:
             self._cancel_scan()
             self._cancel_scan = None
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return unique id."""
+        if not self._unique_id:
+            raise ClassSetupMissing()
```

Comment:
It would be better to refactor and create the connection and connection info before instantiating `FritzBoxTools` and then pass the connection and connection info to `FritzBoxTools` when instantiating it.

Reply:
Can we leave this task for a new PR ? It's quite big change and I have also the switch PR waiting for merging ;-)

Reply:
That's ok.